### PR TITLE
NIFI-5317 - support non-ASCII X-Mailer header

### PR DESCRIPTION
**Pull request by dtrodrigues**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [x] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [x] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [x] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by ottobackwards** [comment]
This looks good, just two comments.

1. If we are going to add support for non-ascii, the test should test non-ascii, can we add that?
2. We now have a pattern of use in two places using with a try catch around setting headers with encoding, maybe we should have a function?

**Comment by dtrodrigues** [comment]
Header setting is now refactored into a function and there are tests for non-ascii headers

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutEmail.java [hunk 1]
```diff
@@ -320,6 +320,15 @@ public void onScheduled(final ProcessContext context) {
         this.attributeNamePattern = attributeNameRegex == null ? null : Pattern.compile(attributeNameRegex);
     }
```
Comment (ottobackwards) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutEmail.java]: We should include the header name in the message

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestPutEmail.java [hunk 2]
```diff
@@ -137,7 +137,7 @@ public void testOutgoingMessage() throws Exception {
     public void testOutgoingMessageWithOptionalProperties() throws Exception {
         // verifies that optional attributes are set on the outgoing Message correctly
```
Comment (ottobackwards) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestPutEmail.java]: I don't see the non-ascii, am I missing something?

Comment (dtrodrigues) [reply nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestPutEmail.java]: The test is for the non-ascii accented 'i' here in the first 'i' of "TestingNífi" on lines 140/167 as well as for the accented 'u' in "Búlk"

**Comment by dtrodrigues** [comment]
I updated the test X-Mailer header to be more explicit that it's non-ASCII

**Comment by ottobackwards** [comment]
Yikes, sorry.

**Comment by ottobackwards** [comment]
+1 LGTM, mvn install and contrib-check pass, use of MineType utility per best practices.

**Comment by pvillard31** [comment]
+1, merging to master, thanks @dtrodrigues @ottobackwards

**Pull request closed without merging**